CapstoneProject/Assets/Scripts/Core/Entity/Abilities/Shield.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Core.Entity;
+using UnityEngine;
+
+public class Shield : MonoBehaviour
+{
+    private float _existTime;
+    private Player _owner;
+    public void Setup(float existTime, Player player)
+    {
+        _existTime = existTime;
+        _owner = player;
+    }
+
+    private void Update()
+    {
+        _existTime -= Time.deltaTime;
+        if (_existTime < 0)
+            Destroy(this.gameObject);
```

Comment:
This works, but just in case you're not aware, there is an overloaded function for Destroy that takes in a time to destroy as second argument, so you can save yourself the extra if check and variable stored and just on Setup call destroy with the time to destroy in. 

If this is something that is going to be used frequently though, it might be just better off disabling and enabling rather than instantiating and destroying over and over to reduce potential memory fragmentation as there's really not a whole lot to this object.

also, rather than update the position each update to the player, could you not just make it a child of the player?